app/assets/javascripts/components/common/overview_stats_tabs.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import OverviewStatsTab from './OverviewStats/overview_stats_tab';
+import NamespaceOverviewStats from './OverviewStats/namespace_overview_stats';
+import WikidataOverviewStats from './wikidata_overview_stats';
+
+import ArticleUtils from '../../utils/article_utils';
+
+
+const OverviewStatsTabs = ({ statistics }) => {
+  const [currentTabId, setCurrentTabId] = useState(0);
+
+  const getTabTitle = (ns_id, wiki) => {
+    const project = wiki.split('.')[1];
+    let ns_title = ArticleUtils.NamespaceIdMapping[ns_id];
+    if (typeof (ns_title) !== 'string') ns_title = ns_title[project];
+    return `${I18n.t(`namespace.${ns_title}`)} (${wiki})`;
+  };
+
+  const getContentTitle = (ns_id, wiki) => {
+    const project = wiki.split('.')[1];
+    let ns_title = ArticleUtils.NamespaceIdMapping[ns_id];
+    if (typeof (ns_title) !== 'string') ns_title = ns_title[project];
+    return `Stats for ${I18n.t(`namespace.${ns_title}`)} (${wiki})`;
+  };
+
+  const onTabChange = (e) => {
+    return setCurrentTabId(Number(e.currentTarget.id));
+  };
+
+  const hasWikidataStats = () => {
+    if (statistics['www.wikidata.org']) return true;
+    return false;
+  };
+
+  const statsDataList = [];
+  // if there are wikidata overview stats, they should be displayed on first tab
```

Comment:
This and the forEach loop are more data manipulation than I'd like to see. If possible, it would be best to do this in a way that doesn't special-case Wikidata. (I don't think it necessarily needs to be first.)

Reply:
Also, I got this conceptual question. For each tab we have a different content rendered by different component. So, is it better to create this components, store and use them (like I am doing now), or is it better to initialize and render them instantaneously when a tab is selected?

I think in our case, the compoents are not that heavy and so they can probably be stored before hand.